recipes/tbb/all/conanfile.py

```diff
@@ -0,0 +1,153 @@
+            arch = "armv7"
+        elif self.settings.arch == "armv8":
+            arch = "aarch64"
+        if self.settings.compiler in ['gcc', 'clang', 'apple-clang']:
+            if str(self.settings.compiler.libcxx) in ['libstdc++', 'libstdc++11']:
+                extra += " stdlib=libstdc++"
+            elif str(self.settings.compiler.libcxx) == 'libc++':
+                extra += " stdlib=libc++"
+            extra += " compiler=gcc" if self.settings.compiler == 'gcc' else " compiler=clang"
+
+            extra += " gcc_version={}".format(str(self.settings.compiler.version))
+        make = tools.get_env("CONAN_MAKE_PROGRAM", tools.which("make") or tools.which('mingw32-make'))
+        if not make:
+            raise ConanInvalidConfiguration("This package needs 'make' in the path to build")
+
+        with tools.chdir(self._source_subfolder):
+            # intentionally not using AutoToolsBuildEnvironment for now - it's broken for clang-cl
+            if self.is_clanglc:
+                add_flag('CFLAGS', '-mrtm')
+                add_flag('CXXFLAGS', '-mrtm')
+
+            targets = self.get_targets()
+            if self.is_msvc:
+                # intentionally not using vcvars for clang-cl yet
+                with tools.vcvars(self.settings):
+                    runtime = "vc14.2"
+                    self.run("%s arch=%s runtime=%s %s %s" % (make, arch, runtime, extra, " ".join(targets)))
+            elif self.is_mingw:
+                self.run("%s arch=%s compiler=gcc %s %s" % (make, arch, extra, " ".join(targets)))
+            else:
+                self.run("%s arch=%s %s %s" % (make, arch, extra, " ".join(targets)))
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="*.h", dst="include", src="%s/include" % self._source_subfolder)
+        self.copy(pattern="*", dst="include/tbb/compat", src="%s/include/tbb/compat" % self._source_subfolder)
+        build_folder = "%s/build/" % self._source_subfolder
+        build_type = "debug" if self.settings.build_type == "Debug" else "release"
+        self.copy(pattern="*%s*.lib" % build_type, dst="lib", src=build_folder, keep_path=False)
+        self.copy(pattern="*%s*.a" % build_type, dst="lib", src=build_folder, keep_path=False)
+        self.copy(pattern="*%s*.dll" % build_type, dst="bin", src=build_folder, keep_path=False)
+        self.copy(pattern="*%s*.dylib" % build_type, dst="lib", src=build_folder, keep_path=False)
+        # Copy also .dlls to lib folder so consumers can link against them directly when using MinGW
+        if self.settings.os == "Windows" and self.settings.compiler == "gcc":
+            self.copy("*%s*.dll" % build_type, dst="lib", src=build_folder, keep_path=False)
+
+        if self.settings.os == "Linux":
+            # leaving the below line in case MacOSX build also produces the same bad libs
+            extension = "dylib" if self.settings.os == "Macos" else "so"
```

Comment:
The following if is never going to be true as it is under `if self.settings.os == "Linux"` clause

Reply:
removed condition